packages/types/src/Credential.ts

```diff
@@ -21,8 +21,14 @@ export interface ICredential {
   claim: IClaim
   claimNonceMap: Record<Hash, string>
   claimHashes: Hash[]
-  claimerSignature?: DidSignature & { challenge?: string }
   delegationId: IDelegationNode['id'] | null
   legitimations: ICredential[]
   rootHash: Hash
 }
+
+export interface ICredentialPresentation extends ICredential {
+  claimerSignature: DidSignature & { challenge?: string }
+}
+
+export type CredentialOrPresentation = ICredential &
+  Partial<ICredentialPresentation>
```

Comment:
Would something like that be more clear?
```suggestion

type ClaimerSignature = DidSignature & { challenge?: string }

export interface ICredentialPresentation extends ICredential {
  claimerSignature: ClaimerSignature
}

export interface CredentialOrPresentation extends ICredential {
  claimerSignature?: ClaimerSignature
}
```

Reply:
Alternatively
```suggestion
interface PresentationProperties {
  claimerSignature: DidSignature & { challenge?: string }
}

export type ICredentialPresentation = ICredential & Required<PresentationProperties>

export type CredentialOrPresentation = ICredential & Partial<PresentationProperties>
```

Reply:
`CredentialOrPresentation` is the result of the one breaking change that the split into credential and presentation could introduce: the message body types and the expectations associated with those. While it is pretty clear that for `submit-credential` we do expect the credential to be signed (i.e. we expect a presentation) it is less clear for `request-attestation`, because we didn't really define whether or not a request for attestation needs to be signed or not. So I see two possible solutions here: 
a) We make a decision here and say that requests for attestations come in the form of a presentation (or a credential). With regards to backwards compatibility this could lead to users of the new sdk having different expectations than users of the old sdk; but even then, this would not be dramatic as one side would take a property to be optional, while the other takes it as mandatory - or simply ignores it.
b) We do not make a decision here but rename `CredentialOrPresentation` to `ICredentialRequest` to make it explicit that the request can come signed or unsigned.

Reply:
I’m fine with both options.

I made the code suggestions because I found `ICredential & Partial<ICredentialPresentation>` a bit confusing, since `ICredentialPresentation` extends `ICredential`.